lib/install_shared.sh

```diff
@@ -282,9 +285,36 @@ function finalize() {
     fi
 }
 
+function generate_unique_hash() {
+    if [[ $(uname) == "Darwin" ]]; then
+        system_profiler SPHardwareDataType | awk '/Serial Number/ {print $4}'
+    else
+        cat /etc/machine-id
+    fi
+}
 
+function send_ping() {
+    local unique_hash=$(generate_unique_hash)
+    local random_hash=$(openssl rand -hex 8)
+    local arch=$(uname -m)
+    local os=$(uname -s)
+    local os_version=$(uname -r)
+
+    curl -i -X POST https://plausible.io/api/event \
+        -H 'User-Agent: ${random_hash}' \
+        -H 'Content-Type: application/json' \
+        --data "{\"name\":\"install\",\"url\":\"http://hello.green-coding.io/install\",\"domain\":\"hello.green-coding.io\",\"props\":{\"unique_hash\":\"${unique_hash}\",\"arch\":\"${arch}\",\"arch\":\"${os}\",\"os\":\"${os}\",\"os_version\":\"${os_version}\"}}" > /dev/null
+}
 
-while getopts "p:a:m:nhtbisyrlc:k:e:" o; do
+function ask_for_ping() {
+    echo ""
+    read -p "Developing software can be a lonely business. Want to let us know you are installing the GMT? No personal data will be shared! (y/N) : " send_ping_input
```

Comment:
@ArneTR read is here

Reply:
I think your code has a bug then. You refactored it to trigger the ask_for_ping function and not send_ping if -Z is set. So it will always ask 

I am in favor of skipping the question with appropriate params. But this is not happening to my understanding atm

Reply:
no, I stand corrected. Was a misunderstanding

Reply:
I think you are right actually. The check is in the wrong place.